### DISCUSSION TITLE
Historycache / History snapshot

### DISCUSSION THREAD

**Discussion by albinlju** [hydrostack/hydro#40] [Ideas]
First and foremost, I want to express my appreciation for the incredible work done on Hydro! Hydro fills the gap for what i wish for in the .net ecosystem. Grateful for the continuous improvements!

I would like to propose a new feature inspired by Htmx, specifically the "history snapshot" functionality. This feature allows the application to take a snapshot of the current state and restore it later, providing a seamless user experience when navigating through the history.

I believe this would be beneficial as it would:

1.  Enhance user experience by providing smooth navigation on browser forward and backward button with scroll restoration.
2.  Improve performance by reducing redundant data loading.

Here’s a brief overview of how Htmx handles "history snapshots" for reference: [https://htmx.org/docs/#history](https://htmx.org/docs/#history)

**Comment by kjeske** [comment]
Hi! Thank you for kind words!
When it comes to the history snapshots, it's indeed a very good idea to add it to Hydro! Hydro keeps the components' state on the page, so using the cached DOM will restore also their state, perfect! I will add it to the issues. Thanks for a nice suggestion!
